### ISSUE TITLE
Dataset and pretrained model links do not work

### ISSUE THREAD

**Issue by maollila** [xuxy09/RawSR#6]
The google drive links do not work for datasets or pretrained models

**Comment by Mayongrui** [comment]
Yeah, I think the link has expired. I will refresh it recently (I think before next Monday). Thanks for your report!

**Comment by leonmakise** [comment]
Waiting for new links!

**Comment by Mayongrui** [comment]
Dear Jiaqi,

Thanks for your interest in our work. Please find the new link for the
shared documents (training/validation/real data and pre-trained weights)
for CVPR. The x4 SR dataset and weights need several days to prepare, and I
will email you the latest link once they are ready.

Please let me know for any further questions.

Link:
https://drive.google.com/drive/folders/115ndMx97ZemzA_vV1Nf0W8gnEMrt32U5?usp=sharing

Yours,
Yongrui

Jiaqi Ma ***@***.***> 于2022年7月14日周四 17:21写道：

**Comment by maollila** [comment]
Thank you for sharing these. If I understand you correctly you will soon provide x4 and x2 pretrained weights as well. 

Br,
Mikko

**Comment by Mayongrui** [comment]
Yep, and actually I should have already shared the x2 pretrained weights in
the previous link. The x4 weights will be released later.

Yours,
Yongrui

maollila ***@***.***> 于2022年7月18日周一 03:29写道：

**Comment by maollila** [comment]
Hi, 

In that case ,just for testing with pretrained weights, the parameters file should be like below? Is some bias missing from checkpoint as error is resulted?
TRAINING = False
TESTING = True
REAL = False

PRETRAINED = True

**Below error message is resulted:**

 Key main_model/branch_1_0/bottle_1_1/biases not found

tensorflow.python.framework.errors_impl.NotFoundError: Key main_model/branch_1_0/bottle_1_1/biases not found in checkpoint
         [[Node: save/RestoreV2 = RestoreV2[dtypes=[DT_INT32, DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT, ..., DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT], _device="/job:localhost/replica:0/task:0/device:CPU:0"](_arg_save/Const_0_0, save/RestoreV2/tensor_names, save/RestoreV2/shape_and_slices)]]
         [[Node: save/RestoreV2/_79 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device_incarnation=1, tensor_name="edge_84_save/RestoreV2", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:GPU:0"]()]]

**Comment by Mayongrui** [comment]
Let me check for it.

Yours,
Yongrui

maollila ***@***.***> 于2022年7月18日周一 23:35写道：

**Comment by Mayongrui** [comment]
Hi,

Would you mind sharing the code you currently have? I tested on my side,
and do not meet the same issue as you reported.

Yours,
Yongrui

maollila ***@***.***> 于2022年7月18日周一 23:35写道：

**Comment by maollila** [comment]
Hi,
[RawSR-master.zip](https://github.com/xuxy09/RawSR/files/9154956/RawSR-master.zip)

ok, It may be then something in environment. Attached are my files. Also you can find my conda environment. Major differences are Python3 and Tensorflow 2 however I think I tried with tf 1.9 too. ckpt files provided in link above where dated 2019. 

br,
-Mikko

**Comment by maollila** [comment]
Tensorflow 1.9 gives same error. I use these for 1.9  and lines with # for TF2

import tensorflow as tf

#import tensorflow.compat.v1 as tf
#tf.disable_v2_behavior()
#import tf_slim as slim
from tensorflow.contrib import slim

**Comment by maollila** [comment]
I wonder if CKPT files were you provided in above link were incorrect as they were dated 2019. I might have redo them as I understood you would for x4?

**Comment by nessotrin** [comment]
Hello. I checked the dataset, it seems incomplete. The checkpoint doesn't match the model in this repository, with many layers missing. Also, the training data doesn't seem to include ground truth or ISP images. Tools to generate the training data mentioned in the paper are not present either. Any chance you could share what is missing ? RawSR is unusable at the moment without it.

**Comment by GZ-T** [comment]
Hello, I found that unable to get the ground truth images and ISP images in the link you provided, could you please upload the missing data again. In addition, in the TrainingSet file why there are 10 .npy files for the same indexed image? Looking forward to your reply.

Example:
a0004-jmac_MG_13840000.npy
a0004-jmac_MG_13840001.npy
a0004-jmac_MG_13840002.npy
a0004-jmac_MG_13840003.npy
a0004-jmac_MG_13840004.npy
a0004-jmac_MG_13840005.npy
a0004-jmac_MG_13840006.npy
a0004-jmac_MG_13840007.npy
a0004-jmac_MG_13840008.npy
a0004-jmac_MG_13840009.npy

**Comment by Mayongrui** [comment]
Hi,

I should have updated the link early this month. Let me double-check on it
first.

And about the questions for multiple files with the same indexed image, it
is because when we generate training samples given one single indexed
image, we sample the motion kernel and noises multiple times. It means that
one ground truth matching 10 raw/ISP images.

Yours,
Yongrui

GZT ***@***.***> 于2023年12月28日周四 22:00写道：

**Comment by GZ-T** [comment]
Thank you very much for your prompt response and look forward to providing the missing data.

**Comment by lemontree893** [comment]
Hello,I unable to get the complete training data from the link provided above, could you please provide a link with the complete data？
